packages/flame/lib/src/game/game_loop.dart

```diff
@@ -2,47 +2,41 @@ import 'package:flutter/scheduler.dart';
 
 class GameLoop {
   void Function(double dt) callback;
-  Duration previous = Duration.zero;
-  late Ticker _ticker;
+  Duration _previous = Duration.zero;
+  late final Ticker _ticker;
 
   GameLoop(this.callback) {
     _ticker = Ticker(_tick);
   }
 
   void _tick(Duration timestamp) {
-    final dt = _computeDeltaT(timestamp);
+    final durationDelta = timestamp - _previous;
+    final dt = durationDelta.inMicroseconds / Duration.microsecondsPerSecond;
+    _previous = timestamp;
     callback(dt);
   }
 
-  double _computeDeltaT(Duration now) {
-    final delta = previous == Duration.zero ? Duration.zero : now - previous;
-    previous = now;
-    return delta.inMicroseconds / Duration.microsecondsPerSecond;
-  }
-
   void start() {
-    _ticker.start();
+    if (!_ticker.isActive) {
+      _ticker.start();
+    }
   }
 
   void stop() {
     _ticker.stop();
+    _previous = Duration.zero;
   }
 
   void dispose() {
     _ticker.dispose();
   }
 
-  void pause() {
```

Comment:
I don't think they were meant to be the same, IIRC @erickzanardo added those to fix a bug or something

Reply:
Yes, I do remember something like this too, but I can remember exactly what but and why, we should hold this until we can remember

Reply:
I also though we added these, but apparently these were added by someone called @erf 20 months ago, maybe they could explain?

Reply:
Here is the PR: https://github.com/flame-engine/flame/pull/302

Reply:
@spydon hi, i took a look now at my PR where i simplified the game loop to use a Ticker. It looks like `start` and `stop` replaced the `scheduleTick` and `unscheduleTick` calls. The only different between those were that `previous` Duration were set to zero in `pause`. It think it would be fine to combine these calls. Not sure if there would be a case where e.g. the user would e.g. want to pause the callbacks, but still make the timer move forward  or something like that.

Reply:
Thanks for the input @erf!

Reply:
No problem! I would perhaps also take a look at the `Ticker` inner workings, and see if you want to expose e.g. `isActive` or `hidden` to `GameLoop`.

Reply:
> The only different between those were that previous Duration were set to zero in pause.

Note that NOT setting duration to zero in stop, would have produced negative `dt` after the subsequent call to `start`, which is clearly a bug. The only reason it wasn't noticed is because `stop` was never used.

Reply:
the bug was: whenever the widget was recreated, since we were using the engine state as the pause/unpause, it would restart the engine. we needed a difference between user manually calling pause (persistent) and engine being paused/unpaused by the framework.

can we test this new version under that scenario?

Reply:
@luanpotter any suggestion for how this test should look?

Reply:
I didn't mean an automated test, just manually really!

Reply:
Interesting... So I tried the following manual test: launch examples/System/Pause-resume-engine, turn on a device frame, click on the helicopter to pause the engine, then click the button to rotate the device frame (which involves rebuilding the widget).

The helicopter starts spinning again, the engine doesn't stay paused. So I guess the test fails? I also tried the same test in the current `main`, and the outcome is the same: the engine restarts, test fails.

It looks like the reason is that the "ticker" of the engine belongs to the `GameLoop` class, which in turn belongs to the `GameRenderBox`, which in turns gets re-created via the `_GameRenderObjectWidget` whenever the `GameWidget` is rebuilt. So, when `GameWidget` is rebuilt, the `GameLoop` gets recreated, and it starts running.

In short, the bug that you describe currently exists in `main`, and this PR neither fixes it nor makes it worse. Should we perhaps open a separate issue for the bug?

Reply:
Yeah, then I think we can just as well merge this and file a bug for that behavior.

Reply:
Can you file an issue for that @st-pasha since you have the most context?

Reply:
Btw, in your test, did you declare the game outside of the `GameWidget`?

Reply:
I've tested with the standard example from our Dashbook, I'll have to check if it recreates the game instance upon frame rotation.

Reply:
Ok, I checked again -- after making sure that the game instance is declared only once in the examples app, the "bug" disappears (i.e. the game state keeps being paused after rotating the frame or changing window size, or opening the side menu). This happens both in the main branch and in this branch.

Reply:
Super, then it works as intended!